Source/ACE.Server/Command/Handlers/AdminCommands.cs

```diff
@@ -1584,59 +1584,268 @@ public static void HandleFumble(Session session, params string[] parameters)
         public static void HandleGod(Session session, params string[] parameters)
         {
             // @god - Sets your own stats to a godly level.
+            // need to save stats so that we can return with /ungod
+            var biotas = new Collection<(Biota biota, ReaderWriterLockSlim rwLock)>();
+            biotas.Add((session.Player.Biota, session.Player.BiotaDatabaseLock));
+            DatabaseManager.Shard.SaveBiotasInParallel(biotas, result => DoGodMode(result, session));
+        }
+            
+        private static void DoGodMode(bool playerSaved, Session session, bool exceptionReturn = false)
+        {
+            if (!playerSaved)
+            {
+                ChatPacket.SendServerMessage(session, "Error saving player. Godmode not available.", ChatMessageType.Broadcast);
+                Console.WriteLine($"Player {session.Player.Name} tried to enter god mode but there was an error saving player. Godmode not available.");
+                return;
+            }
+
+            Biota biota = DatabaseManager.Shard.GetBiota(session.Player.Guid.Full);
+
+            var godString = biota.BiotaPropertiesString.FirstOrDefault(s => s.Type == (ushort)PropertyString.GodState);
+
+            if (!exceptionReturn)
+            {
+                // if godstate starts with 1, you are in godmode
+
+                if (godString != null)
+                {
+                    if (godString.Value.StartsWith("1"))
+                    {
+                        ChatPacket.SendServerMessage(session, "You are already a god.", ChatMessageType.Broadcast);
+                        return;
+                    }
+                }
+
+                string returnState = "1=";
+                returnState += $"{DateTime.UtcNow}=";
+
+                // need level, available skill credits
+                foreach (var i in biota.BiotaPropertiesInt)
+                {
+                    switch (i.Type)
+                    {
+                        case 24:
+                            returnState += $"{i.Type}=";
+                            returnState += $"{i.Value}=";
+                            break;
+                        case 25:
+                            returnState += $"{i.Type}=";
+                            returnState += $"{i.Value}=";
+                            break;
+                        default:
+                            break;
+                    }
+                }
+
+                // need total xp, unassigned xp
+                foreach (var iSixFour in biota.BiotaPropertiesInt64)
+                {
+                    returnState += $"{iSixFour.Type}=";
+                    returnState += $"{iSixFour.Value}=";
+                }
+
+                // need all attributes
+                foreach (var att in biota.BiotaPropertiesAttribute)
```

Comment:
similar, why not just use session.Player.Attributes?

accessing a copy of the raw Biota for the player seems very odd here

Reply:
it was in the format i needed and was easy to loop through this way. i see what you mean though. you are suggesting looping through like i do with the stats increase?
`foreach (var a in currentPlayer.Attributes)
            {
                add values to string here
            }`

how is this different that accessing the "raw" biota as you call it?